File: sandbox/ds/smoke.js
// Require-only smoke test for sandbox-runtime-ds Node deps.
// Run inside the image (NODE_PATH already pre-set in the Dockerfile).
// Exits 0 on success, 1 on the first failed require.

const modules = ["pptxgenjs", "docx"];
const failures = [];
for (const m of modules) {
    try {
        require(m);
    } catch (err) {
        failures.push([m, `${err.name}: ${err.message}`]);
    }
}
if (failures.length > 0) {
    console.error("node-deps: FAIL");
    for (const [m, msg] of failures) console.error(`  ${m}: ${msg}`);
    process.exit(1);
}
console.log(`node-deps: OK (${modules.length} modules)`);

File: sandbox/ds/smoke.py
"""Import-only smoke test for sandbox-runtime-ds.

Run inside the image (root or agent user) to confirm every bundled
Python library imports cleanly. Exits 0 on success, 1 on the first
failed import.

Used by GHA after `docker build` — failure here blocks the release.
"""

import importlib
import sys

# Mirror sandbox/ds/requirements.txt; one entry per top-level module
# the operator should be able to `import` without per-session pip work.
MODULES = [
    # pd-skills + general DS core
    "pandas",
    "numpy",
    "scipy",
    "pyarrow",
    "plotly",
    "matplotlib",
    "seaborn",
    "sklearn",
    "xgboost",
    "catboost",
    "shap",
    "umap",
    "statsmodels",
    "ruptures",
    "mlxtend",
    "numba",
    # Heavyweight optionals
    "dowhy",
    "econml",
    "dask",
    # Data IO
    "openpyxl",
    "sqlalchemy",
    "adbc_driver_postgresql",
    "lxml",
    # Anthropic skills (pptx/xlsx/docx — Python side)
    "markitdown",
    "pptx",
    "docx",
    "PIL",
]


def main() -> int:
    failures: list[tuple[str, str]] = []
    for mod in MODULES:
        try:
            importlib.import_module(mod)
        except Exception as exc:  # noqa: BLE001
            failures.append((mod, f"{type(exc).__name__}: {exc}"))
    if failures:
        print("python-deps: FAIL", file=sys.stderr)
        for mod, msg in failures:
            print(f"  {mod}: {msg}", file=sys.stderr)
        return 1
    print(f"python-deps: OK ({len(MODULES)} modules)")
    return 0


if __name__ == "__main__":
    sys.exit(main())
